doc/develop/application/index.rst

```diff
@@ -67,12 +67,103 @@ generated in a build directory; Zephyr does not support "in-tree" builds.
 The following sections describe how to create, build, and run Zephyr
 applications, followed by more detailed reference material.
 
-Example standalone application
+.. _zephyr-app-types:
+
+Application types
+*****************
+
+Based on where the source code of the application is located we can distinguish
+between three basic application types.
+
+* Zephyr repository application
+* Zephyr workspace application
+* Zephyr freestanding application
+
+You can find out more about how the build system supports all the application
+types described in this section in the :ref:`cmake_pkg` section.
+
+.. _zephyr-repo-app:
+
+Zephyr repository application
+=============================
+
+An application located within the ``zephyr`` folder in a Zephyr :ref:`west
+workspace <west-workspaces>` is referred to as a Zephyr repository application.
+In the following example, the :ref:`hello_world sample <hello_world>` is a
+Zephyr repository application:
+
+.. code-block:: none
+
+   zephyrproject/
+   ├─── .west/
+   │    └─── config
+   └─── zephyr/
+        ├── arch/
+        ├── boards/
+        ├── cmake/
+        ├── samples/
+        │    ├── hello_world/
+        │    └── ...
+        ├── tests/
+        └── ...
+
+.. _zephyr-workspace-app:
+
+Zephyr workspace application
+============================
+
+An application located within a :ref:`workspace <west-workspaces>`, but outside
+the Zephyr repository (and thus folder) itself, is referred to as a Zephyr
+workspace application.  In the following example, ``app`` is a Zephyr workspace
+application:
```

Comment:
I'd add a note here mentioning this is likely the preferred way to develop Zephyr applications so that it is clear for users. I've seen quite a few different application structures, probably because documentation is not clear nor doesn't point users to the common use case.

Reply:
See my earlier comments about this:

https://github.com/zephyrproject-rtos/zephyr/pull/44803#discussion_r874980951
https://github.com/zephyrproject-rtos/zephyr/pull/44803#discussion_r874984831